generic3g/specs/FrequencyAspect.F90

```diff
@@ -55,9 +55,9 @@ function new_FrequencyAspect(timeStep, refTime, accumulation_type) result(aspect
       call aspect%set_time_dependent(.FALSE.)
       call aspect%set_accumulation_type(INSTANTANEOUS)
       call aspect%zero_timestep()
-      if(present(timestep)) aspect%timestep_ = timestep
-      if(present(refTime)) aspect%refTime_ = refTime
-      if(present(accumulation_type)) aspect%accumulation_type_ = accumulation_type
+      if(present(timestep)) call aspect%set_timestep(timestep)
+      if(present(refTime)) call aspect%set_reference_time(refTime)
+      if(present(accumulation_type)) call aspect%set_accumulation_type(accumulation_type)
```

Comment:
I do not understand the verbosity here.  The other setters (e.g., `set_mirror()`) are necessary because the corresponding data component is PRIVATE in the base type. 

```suggestion
      if(present(timestep)) aspect%timeStep = timeStep
      if(present(refTime)) aspect%refTime = refTime
      if(present(accumulation_type)) aspect%accumulation_type = accumulation_type
```

Reply:
I made the change for time step and reference time. The accumulation type cannot be set simply. The setter checks the actual accumulation type to make sure it is a valid accumulation type, and it 

The setter updates the instance accumulation type only if the actual accumulation type argument if it is valid, so that the constructor does not return an instance with an invalid accumulation type.

Reply:
Got it.

So you'll also remove the 2 setter procedures, right?